### ISSUE TITLE
Bug running project-config/apply after deleting FeedMe

### ISSUE THREAD

**Issue by michtio** [verbb/navigation#224]
**Description**

Running into the issue when locally uninstalling FeedMe and having the Navigation plugin enabled.
So after having FeedMe installed together with the Navigation plugin.

**Steps to reproduce**
1. Have both Navigation / FeedMe installed
2. Go to plugins->settings --> uninstall FeedMe
3. Remove FeedMe from composer
4. Deploy to server
5. Updated composer runs / feedme gets removed / not installed.
6. Try to run project-config/apply --> it errors because in the database Navigation still relies on FeedMe, but because composer doesn't install it, it can't be found, and gives the following error:

```
Exception 'Error' with message 'Class 'craft\feedme\services\Elements' not found'
in /home/forge/sandbox.percipio.london/releases/87186c3989d7a3a706b9f590d9bfbb74a0fb09ef/vendor/verbb/navigation/src/Navigation.php:216
Stack trace:
#0 /home/forge/sandbox.percipio.london/releases/87186c3989d7a3a706b9f590d9bfbb74a0fb09ef/vendor/verbb/navigation/src/Navigation.php(75): verbb\navigation\Navigation->_registerFeedMeSupport()
#1 /home/forge/sandbox.percipio.london/releases/87186c3989d7a3a706b9f590d9bfbb74a0fb09ef/vendor/yiisoft/yii2/base/BaseObject.php(109): verbb\navigation\Navigation->init()
#2 /home/forge/sandbox.percipio.london/releases/87186c3989d7a3a706b9f590d9bfbb74a0fb09ef/vendor/yiisoft/yii2/base/Module.php(158): yii\base\BaseObject->__construct()
#3 /home/forge/sandbox.percipio.london/releases/87186c3989d7a3a706b9f590d9bfbb74a0fb09ef/vendor/craftcms/cms/src/base/Plugin.php(115): yii\base\Module->__construct()
#4 [internal function]: craft\base\Plugin->__construct()
#5 /home/forge/sandbox.percipio.london/releases/87186c3989d7a3a706b9f590d9bfbb74a0fb09ef/vendor/yiisoft/yii2/di/Container.php(406): ReflectionClass->newInstanceArgs()
#6 /home/forge/sandbox.percipio.london/releases/87186c3989d7a3a706b9f590d9bfbb74a0fb09ef/vendor/yiisoft/yii2/di/Container.php(165): yii\di\Container->build()
#7 /home/forge/sandbox.percipio.london/releases/87186c3989d7a3a706b9f590d9bfbb74a0fb09ef/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get()
#8 /home/forge/sandbox.percipio.london/releases/87186c3989d7a3a706b9f590d9bfbb74a0fb09ef/vendor/craftcms/cms/src/services/Plugins.php(909): yii\BaseYii::createObject()
#9 /home/forge/sandbox.percipio.london/releases/87186c3989d7a3a706b9f590d9bfbb74a0fb09ef/vendor/craftcms/cms/src/services/Plugins.php(228): craft\services\Plugins->createPlugin()
#10 /home/forge/sandbox.percipio.london/releases/87186c3989d7a3a706b9f590d9bfbb74a0fb09ef/vendor/craftcms/cms/src/base/ApplicationTrait.php(1416): craft\services\Plugins->loadPlugins()
#11 /home/forge/sandbox.percipio.london/releases/87186c3989d7a3a706b9f590d9bfbb74a0fb09ef/vendor/craftcms/cms/src/console/Application.php(48): craft\console\Application->_postInit()
#12 /home/forge/sandbox.percipio.london/releases/87186c3989d7a3a706b9f590d9bfbb74a0fb09ef/vendor/yiisoft/yii2/base/BaseObject.php(109): craft\console\Application->init()
```

The culprit would be in here:

```
 private function _registerFeedMeSupport()
    {
        $feedMe = Craft::$app->getPlugins()->isPluginInstalled('feed-me') && Craft::$app->getPlugins()->isPluginEnabled('feed-me');
        if ($feedMe) {
            Event::on(FeedMeElements::class, FeedMeElements::EVENT_REGISTER_FEED_ME_ELEMENTS, function(RegisterFeedMeElementsEvent $e) {
                $e->elements[] = NodeFeedMeElement::class;
            });
        }
    }
```

When going into the server and in the vendor folder adjusting the above to:

```
private function _registerFeedMeSupport()
    {
        $feedMe = Craft::$app->getPlugins()->isPluginInstalled('feed-me') && Craft::$app->getPlugins()->isPluginEnabled('feed-me');
    }
```

so basically bypassing/removing the check if feedMe is installed, then project.yaml does work and the changes get applied without issues.

**Additional info**
- Plugin version: latest
- Craft version: 3.5.16
- Multi-site: No

**Comment by engram-design** [comment]
Thanks for investigating this. A few people have come across this issue, which I thought initially might be an isolated incident, but seems it's affecting a few.

What I don't quite understand is that the presence of `use craft\feedme\services\Elements as FeedMeElements;` shouldn't cause an issue, as its only when using `FeedMeElements` that this becomes an issue, which is _supposed_ to be protected around the conditional the plugin is installed. I suppose it's too late to see what `$feedMe` would've evaluated to?

I'll add some checks (`if (class_exists(FeedMeElements::class))`) shortly to improve this.

**Comment by engram-design** [comment]
Fixed in [1.4.5](https://github.com/verbb/navigation/releases/tag/1.4.5)

**Comment by michtio** [comment]
@engram-design I can always leave 1.4.4 installed and investigate further if you'd like, seems like it might also affect the FeedMe plugin itself.

But your check is mainly if feedme is installed and enabled. So according to the database it is installed and enabled, as project.yaml is supposed to throw it out, but it won't because once trying to run the `project-config/apply` it immediately throws that error. 

So I think the code fetches from the database and says "Oh yes! It is installed and enabled!" I don't have any issue when I uninstall it first, and then remove it from composer.json ( as with the action my local database updates , so the navigation plugin knows it's not there anymore ) 

But when deploying to the server, where we disabled allow_admin_changes we can never uninstall through the settings. Triggering the error.

That's as far as I could get :)

**Comment by engram-design** [comment]
All good, should be fixed properly in the next version!
